proposals/0000-merge-datepickerios-and-datepickerandroid.md

```diff
@@ -0,0 +1,229 @@
+> `DatePickerAndroid.dismissedAction` and all the other keys being undefined.
+
+Both platforms require `props.date` to be set to select a date other than the current. i.e. there is no 
+difference in using the default date or setting a specific date from state. State is also not retained on
+Android when the calendar is dismissed.
+
+The default `mode` is `datetime` for iOS. This default is [specified in JS][datetime] through `defaultProps`. 
+Obviously this `mode` is [not available for android][open-android] as it depends on separate Components. 
+The invocation of `DatePickerAndroid` and `TimePickerAndroid` is very similar. The default `mode` for both components 
+is `default`, which matches the default implementation of the running Android version. 
+
+Finally, iOS allows `date` to be a `timestamp` or `Date` object. `DatePickerAndroid` matches this 
+implementation. The `TimePickerAndroid` however requires a `number` to be passed to the `hour` and `minute` props. 
+
+---
+
+## Detailed design
+
+Deprecate the existing modules `DatePickerIOS`, `DatePickerAndroid` and `TimePickerAndroid` as outlined 
+by the ['Lean Core' proposal][lean]. The modules will be replaced with a new single Component 
+called `DateTimePicker`. Which will support a set of properties that cover configurability for both Android
+and iOS. A simple example implementation would look like: 
+
+```js
+class Calendar extends React.Component {
+  state = {
+    date: new Date()
+  };
+
+  render() {
+    return <DateTimePicker
+      value={ this.state.date }                             
+      onDateChange={ date => this.setState({ date }) }      // date is `Date` object
+      mode='date'                                           // new default
+    />
+  }
+}
+```
+
+This requires a new component called `DateTimePickerAndroid` to be created. It will effectively merge 
+the API's of `DatePickerAndroid` and `TimePickerAndroid`. It wraps the function calls required for Android to create 
+a React component. It could be a functional component, e.g.
+
+```js
+function DateTimePickerAndroid({ mode, onDateChange, value, ...props }) {
+  const Selector = mode === 'time' ? TimePickerAndroid : DatePickerAndroid;
+
+  Selector.open({ ...value, ...props }) // act on thenable with `onDateChange`
+}
+```
+
+### Naming
+
+The proposed name is `react-native-datetimepicker`, with a Github repo in the react native community 
+`git@github.com:react-native-community/react-native-datetimepicker.git`. However, suggestions are welcome!
+
+### Options 
+
+Property names should remain similar where possible. However there are some difference between android and iOS
+that need to be resolved. This section will talk about required changes.
+
+#### value
+
+Both `DatePickerIOS` and `DatePickerAndroid` use `date` whereas `TimePickerAndroid` requires an `hour` and 
+`minute` property. The proposal is to change this property to `value {Date, Number}`. Where `{Number}` is
```

Comment:
@mmcknett From above:

> Finally, iOS allows `date` to be a `timestamp` or `Date` object. `DatePickerAndroid` matches this 
implementation. The `TimePickerAndroid` however requires a `number` to be passed to the `hour` and `minute` props. 

I think that answers it?

Reply:
Ah, got it, it's the same as `date` in the iOS version, but with additional genericness in the name so that it can be used to pass a countdown value (as described below) as well. Might be worth clarifying that in "Where `{Number}` is..."

Reply:
Yeah, I was looking for the best way to convey this, but glad the additional comments make it clear. If you want me to change this to some other way to denote a `Set` of options, I'll change it :)